datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -131,32 +132,27 @@ impl std::fmt::Display for BinaryExpr {
     }
 }
 
-/// Invoke a compute kernel on a pair of binary data arrays
```

Comment:
this macro formulation I think really obscures what is going on and actually is more verbose than just making a function that calls concat_elements directly, so I did